examples/flutter_gallery/lib/demo/customized/home.dart

```diff
@@ -0,0 +1,151 @@
+  final GlobalKey<RunStatsContentState> _statsContentKey = GlobalKey<RunStatsContentState>();
+  Widget _statsContentWidget;
+  final GlobalKey<InitialWelcomeState> _initialWelcomeKey = GlobalKey<InitialWelcomeState>();
+  Widget _initialContentWidget;
+  final ScrollController _scrollController = ScrollController();
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _configureThemes();
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _initialContentWidget ??= InitialWelcome(key: _initialWelcomeKey);
+    _statsContentWidget ??= RunStatsContent(
+      key: _statsContentKey,
+      peekAmount: _kStatsSectionPeekAmount,
+      onHeaderTapped: tappedStatsDetailBar,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _mainContentWidget ??= _contentWidget();
+    return WillPopScope(
+      onWillPop: () {
+        if (_animatedStats) {
+          _scrollController.animateTo(
+            0.0,
+            curve: Curves.easeOut,
+            duration: const Duration(milliseconds: 300),
+          );
+        }
+        return Future<bool>.value(!_animatedStats);
+      },
+      child: Theme(
+        data: _themeData,
+        child: Material(
+          color: const Color(0x00FFFFFF),
+          child: _mainContentWidget,
+        ),
+      ),
+    );
+  }
+
+  void _configureThemes() {
+    _themeData = ThemeData(
+      primaryColor: const Color(0xFF212024),
+      buttonColor: const Color(0xFF3D3D3D),
+      iconTheme: const IconThemeData(color: Color(0xFF4A4A4A)),
+      brightness: Brightness.light,
+    );
+  }
+
+  Widget _contentWidget() {
+    final double screenHeight = MediaQuery.of(context).size.height;
+    final String backTitle = Platform.isAndroid ? 'TrackFit' : '';
+    return Scaffold(
+      backgroundColor: const Color(0xFF212024),
+      body: GlowingOverscrollIndicator(
+        color: const Color(0x00FFFFFF),
+        axisDirection: AxisDirection.down,
+        showLeading: false,
+        showTrailing: false,
+        child: NotificationListener<ScrollNotification>(
+          onNotification: _handleScrollNotification,
+          child: CustomScrollView(
+            controller: _scrollController,
+            physics: SnappingScrollPhysics(midScrollOffset: screenHeight),
+            shrinkWrap: true,
+            slivers: <Widget>[
+              SliverAppBar(
+                pinned: false,
+                title: Text(backTitle),
+                expandedHeight: screenHeight - _kStatsSectionPeekAmount - MediaQuery.of(context).padding.top,
+                leading: const BackButton(
+                  color: Colors.white,
+                ),
+                backgroundColor: const Color(0xFF212024),
+                flexibleSpace: FlexibleSpaceBar(
+                  background: _initialContentWidget,
+                ),
+              ),
+              SliverToBoxAdapter(
+                child: _statsContentWidget,
```

Comment:
You probably don't need to cache these instances. The State is kept as long as you don't change the tree shape.

Reply:
Moved to local